### PR TITLE
Update OtaPalWriteBlock_t comment with a note

### PR DESCRIPTION
**Pull request by pvyawaha**
Update OtaPalWriteBlock_t comment with a note about returning from the function and exception condition. 

Description
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is formatted using Uncrustify.
- [x] I have read and applied the rules stated in CONTRIBUTING.md. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by dachalco — DISMISSED** [review]

File: source/include/ota_platform_interface.h [hunk 1]
```diff
@@ -192,6 +192,10 @@ typedef OtaPalStatus_t ( * OtaPalCloseFile_t )( OtaFileContext_t * const pFileCo
 /**
  * @brief Write a block of data to the specified file at the given offset.
  *
+ * Note: This function is supposed to write a complete file block before returning. So the number of
```
Comment (paulbartell) [inline source/include/ota_platform_interface.h:195]:
```suggestion
 * Note: This function must always write a complete file block before returning. So the number of
```

File: source/include/ota_platform_interface.h [hunk 2]
```diff
@@ -192,6 +192,10 @@ typedef OtaPalStatus_t ( * OtaPalCloseFile_t )( OtaFileContext_t * const pFileCo
...
+ * bytes written will be equal to the size of the file block. Exception only in case of last block
+ * where the number of bytes written will be equal or less than the size of the file block.
+ *
```
Comment (paulbartell) [inline source/include/ota_platform_interface.h:198]: Shouldn't the caller verify this condition with an assertion?

Comment (ActoryOu) [reply source/include/ota_platform_interface.h:198]: Hi Paul,
Updated at #435. Trigger assertion at this condition.

Thanks.

**Comment by ActoryOu** [comment]
Updated at #436 . Close this PR.

**Pull request closed without merging**